GitHubComments.ps1

```diff
@@ -0,0 +1,545 @@
+
+        Deletes a Github comment from the PowerShell\PowerShellForGitHub project.
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParametersetName='Elements')]
+    [Alias('Delete-GitHubComment')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(ParameterSetName='Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName='Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName='Uri')]
+        [string] $Uri,
+
+        [Parameter(Mandatory)]
+        [string] $CommentID,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+        'CommentID' =  (Get-PiiSafeString -PlainText $CommentID)
+    }
+
+    $params = @{
+        'UriFragment' = "repos/$OwnerName/$RepositoryName/issues/comments/$CommentID"
+        'Method' = 'Delete'
+        'Description' =  "Removing comment $CommentID for $RepositoryName"
+        'AccessToken' = $AccessToken
+        'TelemetryEventName' = $MyInvocation.MyCommand.Name
+        'TelemetryProperties' = $telemetryProperties
+        'NoStatus' = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus -ConfigValueName DefaultNoStatus)
+    }
+
+    return Invoke-GHRestMethod @params
+}
+
+function Get-CommentAcceptHeader
+{
+<#
+    .DESCRIPTION
+        Returns a formatted AcceptHeader based on the requested MediaType
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER MediaType
+        The format in which the API will return the body of the comment.
+
+        raw - Return the raw markdown body. Response will include body. This is the default if you do not pass any specific media type.
+        text - Return a text only representation of the markdown body. Response will include body_text.
+        html - Return HTML rendered from the body's markdown. Response will include body_html.
+        full - Return raw, text and HTML representations. Response will include body, body_text, and body_html.
+
+    .EXAMPLE
+        Get-CommentAcceptHeader -MediaType raw
+
+        Returns a formatted AcceptHeader for v3 of the response object
+#>
+    [CmdletBinding()]
+    param(
+        [ValidateSet('raw', 'text', 'html', 'full')]
+        [string] $MediaType ='raw'
+    )
+
+    $acceptHeaders = @('application/vnd.github.squirrel-girl-preview')
+
+    if ('raw' -eq $MediaType)
```

Comment:
I think you can simplify this whole block to just this (given that you know it will always have a value, and your valid types are equivalent to those used in the header):

```powershell
 $acceptHeaders += "application/vnd.github.$script:MediaTypeVersion.$MediaType+json"
```

Reply:
Good idea